dataprofiler/data_readers/graph_data.py

```diff
@@ -5,15 +5,73 @@
 from .csv_data import CSVData
+from . import data_utils
 from .filepath_or_buffer import FileOrBufferHandler
 
 class GraphData(BaseData):
+    """
+    GraphData class to identify, read, and load graph data
+    """
+    data_type = 'graph'
         
-    def __init__(self, input_file_path=None, data=None, options=None):
+    def __init__(self, input_file_path=None, options=None, data=None):
+        """
+        Data class for identifying, reading, and loading graph data. Current 
+        implementation only accepts file path as input. An options parameter is
+        also passed in to specify properties of the input file.
+
+        Possible Options::
+
+            options = dict(
+                delimiter= type: str
+                column_names= type: list(str)
+                source_node= type: int
+                destination_node= type: int
+                target_keywords= type: list(str)
+                source_keywords= type: list(str)
+                header= type: any
+                quotechar= type: str
+            )
+
+        delimiter: delimiter used to decipher the csv input file
+        column_names: list of column names of the csv
+        source_node: index of the source node column, range of (0,n-1)
+        target_node: index of the target node column, range of (0,n-1)
+        target_keywords: list of keywords to automatically identify target/destination node column
+        source_keywords: list of keywords to automatically identify source node column
+        header: location o the header in the file
+        quotechar: quote character used in the delimited file
+
+        :param input_file_path: path to the file being loaded or None
+        :type input_file_path: str
+        :param data: data being loaded into the class instead of an input file
+        :type data: multiple types
+        :param options: options pertaining to the data type
+        :type options: dict
+        :return: None
+        """
+
         BaseData.__init__(self, input_file_path, data, options)
+
+        if data is not None:
+            raise NotImplementedError("Inputting data is not yet implemented.")
+        if input_file_path is None:
+            raise ValueError("Please input a file dataset.")
         if options is None:
             options = dict()
 
+        self._source_node = options.get("source_node", None)
+        self._destination_node = options.get("destination_node", None)
+        self._target_keywords = options.get("target_keywords", ['target', 'destination', 'dst'])
+        self._source_keywords = options.get("source_keywords", ['source', 'src', 'origin'])
+        self._column_names = options.get("column_names", self.csv_column_names(self.input_file_path, self.options))
+
+        self._delimiter = options.get("delimiter", None)
+        self._quotechar = options.get("quotechar", None)
+        self._header = options.get("header", 'auto')
+
+        self._data = self._format_data_networkx()
```

Comment:
Remove from here as this should occur if someone references the data auto. We only want to load the data if we need to load it :)